sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -72,13 +74,17 @@ class StreamExecution(
   /**
    * Tracks how much data we have processed and committed to the sink or state store from each
    * input source.
+   * Only the scheduler thread should modify this field, and only in atomic steps. Other threads
+   * must create a local copy before iterating over this data structure.
```

Comment:
What I meant to say is that two accesses to this field from a thread other than the scheduler thread may return different different values. The functions that access `committedOffsets` from other threads either should access the field only once (`toDebugString`); or they should make a copy (as in the code that consumes `StreamingQueryStatus` objects). Updated the comment in my local copy to something that is hopefully more clear: "Other threads should make a shallow copy if they are going to access this field more than once, since the field's value may change at any time."